pkg/config/loader.go

```diff
@@ -105,36 +125,74 @@ func ParseWinPDTargets(log logr.Logger, vpcCniConfigMap *v1.ConfigMap) (warmIPTa
 		warmPrefixTargetStr, foundWarmPrefix = vpcCniConfigMap.Data[WinWarmPrefixTarget]
 	}
 
-	// If no configuration is found, return 0
-	if !foundWarmIP && !foundMinIP && !foundWarmPrefix {
-		return warmIPTarget, minIPTarget, warmPrefixTarget
-	}
-
+	// If warm IP target config value is not found, or there is an error parsing it, the value will be set to zero
 	if foundWarmIP {
 		warmIPTargetInt, err := strconv.Atoi(warmIPTargetStr)
 		if err != nil {
-			log.Error(err, "failed to parse warm ip target", "warm ip target", warmIPTargetStr)
+			log.V(1).Info("could not parse warm ip target, defaulting to zero", "warm ip target", warmIPTargetStr)
+			warmIPTarget = 0
 		} else {
 			warmIPTarget = warmIPTargetInt
+
+			// Handle secondary IP mode scenario where WarmIPTarget is explicitly configured to zero
+			// In such a case there must always be 1 warm IP to ensure that the warmpool is never empty
+			if !isPDEnabled && warmIPTarget == 0 {
+				log.V(1).Info("Explicitly setting WarmIPTarget zero value not supported in secondary IP mode, will override with 1")
+				warmIPTarget = 1
+			}
 		}
+	} else {
+		log.V(1).Info("could not find warm ip target in ConfigMap, defaulting to zero")
+		warmIPTarget = 0
 	}
+
+	// If min IP target config value is not found, or there is an error parsing it, the value will be set to zero
 	if foundMinIP {
 		minIPTargetInt, err := strconv.Atoi(minIPTargetStr)
 		if err != nil {
-			log.Error(err, "failed to parse minimum ip target", "minimum ip target", minIPTargetStr)
+			log.V(1).Info("could not parse minimum ip target, defaulting to zero", "minimum ip target", minIPTargetStr)
+			minIPTarget = 0
 		} else {
 			minIPTarget = minIPTargetInt
 		}
+	} else {
+		log.V(1).Info("could not find minimum ip target in ConfigMap, defaulting to zero")
+		minIPTarget = 0
 	}
-	if foundWarmPrefix {
+
+	// If PD is enabled and warm prefix target config value is not found, or there is an error parsing it, the value will be set to zero
+	if !isPDEnabled && foundWarmPrefix {
```

Comment:
Look like this can be simplified to 
```
needUpdate := isPDEnabled && foundWarmPrefix
if needUpdate {
    warmPrefixTarget = 0
} else {
    if warmPrefixTargetInt, err := strconv.Atoi(warmPrefixTargetStr); err == nil {
        warmPrefixTarget = warmPrefixTargetInt
    } else {
        warmPrefixTarget = 0
    }
}
```

Reply:
Agreed, your code suggestion is more concise. However it will impact the ability to add the log statements that currently exist. Based on your response to thread https://github.com/aws/amazon-vpc-resource-controller-k8s/pull/443#discussion_r1690063506, I could make the suggested change.

Reply:
I agree this can be further simplified, maybe like
```
warmPrefixTarget = 0
if isPDEnabled && foundWarmPrefix {
  if warmPrefixTargetInt, err := strconv.Atoi(warmPrefixTargetStr); err == nil {
    warmPrefixTarget = warmPrefixTargetInt
  }
}
```
I'm not convinced we need to have a log for each scenario. We can add a comment at the top and `log.Info` at the end of the function to print the values of each config variable. Let me know what you think.

Reply:
I decided to accept this feedback. Acknowledged on not needing to have a log for each scenario. Resolving code after recent code push resolves this.